src/nnet3bin/nnet3-compute.cc

```diff
@@ -0,0 +1,166 @@
+// nnet3bin/nnet3-compute.cc
+
+// Copyright 2012-2015   Johns Hopkins University (author: Daniel Povey)
+//                2015   Vimal Manohar
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "nnet3/nnet-simple-computer.h"
+#include "base/timer.h"
+#include "nnet3/nnet-utils.h"
+
+
+int main(int argc, char *argv[]) {
+  // note: making this program work with GPUs is as simple as initializing the
+  // device, but it probably won't make a huge difference in speed for typical
+  // setups.
+  try {
+    using namespace kaldi;
+    using namespace kaldi::nnet3;
+    typedef kaldi::int32 int32;
+
+    const char *usage =
+        "Propagate through neural network model\n"
+        "Usage: nnet3-compute [options] <nnet-in> <features-rspecifier> <matrix-wspecifier>\n";
```

Comment:
<nnet-in> to <raw-nnet-in>

On Tue, Oct 6, 2015 at 1:36 AM, Vimal Manohar <notifications@github.com>
wrote:

> In src/nnet3bin/nnet3-compute.cc
> <https://github.com/kaldi-asr/kaldi/pull/212#discussion_r41227324>:
>
> > +#include "base/timer.h"
> > +#include "nnet3/nnet-utils.h"
> > +
> > +
> > +int main(int argc, char *argv[]) {
> > +  // note: making this program work with GPUs is as simple as initializing the
> > +  // device, but it probably won't make a huge difference in speed for typical
> > +  // setups.
> > +  try {
> > +    using namespace kaldi;
> > +    using namespace kaldi::nnet3;
> > +    typedef kaldi::int32 int32;
> > +
> > +    const char *usage =
> > +        "Propagate through neural network model\n"
> > +        "Usage: nnet3-compute [options] <nnet-in> <features-rspecifier> <matrix-wspecifier>\n";
>
> I didn't understand. What is the change I should make?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/212/files#r41227324>.
>